From 9a2b057efe24a5c223e7c788f0b0d732b196aa03 Mon Sep 17 00:00:00 2001
From: Sultan Alsawaf <sultan@kerneltoast.com>
Date: Sun, 5 Jul 2020 00:21:22 -0700
Subject: [PATCH 17/18] i2c: designware: Allow SMBus block reads up to 255
 bytes in length

According to the SMBus 3.0 protocol specification, block transfer limits
were increased from 32 bytes to 255 bytes. Remove the obsolete 32-byte
limitation.

Signed-off-by: Sultan Alsawaf <sultan@kerneltoast.com>
---
 drivers/i2c/busses/i2c-designware-master.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/i2c/busses/i2c-designware-master.c b/drivers/i2c/busses/i2c-designware-master.c
index 83906bb97..00abc02c7 100644
--- a/drivers/i2c/busses/i2c-designware-master.c
+++ b/drivers/i2c/busses/i2c-designware-master.c
@@ -396,7 +396,7 @@ i2c_dw_read(struct dw_i2c_dev *dev)
 			*buf = dw_readl(dev, DW_IC_DATA_CMD);
 			if (flags & I2C_M_RECV_LEN) {
 				/* Ensure length byte is a valid value */
-				if (*buf <= I2C_SMBUS_BLOCK_MAX && *buf > 0)
+				if (*buf > 0)
 					len = i2c_dw_recv_len(dev, *buf);
 				else
 					len = i2c_dw_recv_len(dev, len);
-- 
2.27.0.203.gf402ea6816

